In [1]:
import numpy as np

In [4]:
data =np.genfromtxt( r"C:\Users\ALEX\Downloads\Sample_data.csv",delimiter=',',skip_header=1)

In [5]:
data

array([[ 0., 25., 40.],
       [ 1., 30., 35.],
       [ 2., 28., 38.],
       [ 3., 32., 41.],
       [ 4., 26., 37.],
       [ 5., 29., 36.],
       [ 6., 27., 39.],
       [ 7., 33., 42.],
       [ 8., 31., 38.],
       [ 9., 28., 37.],
       [10., 30., 40.],
       [11., 29., 36.]])

# Measures of Central Tendency

### 1.Calculate the mean sales for Product A and Product B. Compare the average monthly sales for the two products and discuss any observed trends.


In [16]:
average_sales_productA = data[:,1].mean()
average_sales_productB = data[:,2].mean()

In [17]:
average_sales_productA  


29.0

In [18]:
average_sales_productB

38.25

### Product B has higher average monthly sales compared to Product A.

### 2.Find the median sales for each product and discuss how it reflects the central tendency of the sales distribution.

In [23]:
median_sales_productA = np.median(data[:, 1])
median_sales_productB = np.median(data[:, 2])


In [24]:
median_sales_productA  


29.0

In [25]:
median_sales_productB 


38.0

### The median sales for each product reflect the central tendency of the sales distribution, indicating that half of ### the months had sales above this number and half below. The median sales for Product A and Product B are ### consistent with their respective average sales

### 3.Identify the mode of the sales data for both Product A and Product B. Discuss the significance of the mode in the context of these products.

In [48]:
import statistics

Product_A = data[:,1]
Product_B = data[:,2]

a = statistics.mode(Product_A)
b = statistics.mode(Product_B)

print('Mode Sale for Product A : ', a)
print('Mode Sale for Product B : ', b)

Mode Sale for Product A :  30.0
Mode Sale for Product B :  40.0


#### For both Product A and Product B, there are multiple sales values that occur with the same highest frequency.
#### The presence of multiple modes can suggest that the sales data may have several peaks in its distribution 

### 4.Calculate the weighted mean of the sales data, assuming each month has an equal weight. Discuss any implications of using weighted mean in this context.

In [31]:
weighted_mean_productA = data[:,1].mean()
weighted_mean_productB = data[:,2].mean()


print('Weighted mean for Product A:', weighted_mean_productA)
print('Weighted mean for Product B:', weighted_mean_productB)

Weighted mean for Product A: 29.0
Weighted mean for Product B: 38.25


#### where each month has an equal weight, is the same as the arithmetic mean. It represents the average sales per month for each product when all months are considered equally important

# Measures of Dispersion

### 1.Compute the range of monthly sales for both Product A and Product B. Discuss how the range provides insights into the variability of sales.

In [33]:
productA = data[:,1].max() - data[:,1].min()
productB = data[:,2].max() - data[:,2].min()

print('Range of monthly sales for Product A:', productA)
print('Range of monthly sales for Product B:', productB)

Range of monthly sales for Product A: 8.0
Range of monthly sales for Product B: 7.0


### 2.Calculate the variance of the sales data for both products. Discuss what the variance reveals about the spread of sales for each product


In [34]:
productA = data[:,1].var()
productB = data[:,2].var()

print('Variance of sales for Product A:', productA)
print('Variance of sales for Product B:', productB)

Variance of sales for Product A: 5.166666666666667
Variance of sales for Product B: 4.354166666666667


### Product A has a slightly higher variance than Product B

### 3.Determine the standard deviation of the sales for Product A and Product B. Discuss the significance of standard deviation in understanding the variability in sales.

In [35]:
productA = data[:,1].std()
productB = data[:,2].std()

print('Standard deviation of sales for Product A:', productA)
print('Standard deviation of sales for Product B:', productB)

Standard deviation of sales for Product A: 2.273030282830976
Standard deviation of sales for Product B: 2.0866640042581523


### Product A has a higher standard deviation compared to Product B. showing greater variability in its monthly sales figures

# Relative Measures of Dispersion

### 1.Calculate the coefficient of variation for the sales data of Product A. Discuss how this relative measure of dispersion helps in comparing the variability of products with different average sales.

In [38]:
mean_productA = data[:,1].mean()
std_productA = data[:,1].std()
coefficient_of_variation_productA = (std_productA / mean_productA)*100

print('Coefficient of variation for Product A:', coefficient_of_variation_productA)

Coefficient of variation for Product A: 7.838035458037848


### 2.Compute the relative range for Product B. Discuss how the relative range accounts for the scale of the sales data.


In [39]:
max_productB = data[:,2].max()
min_productB = data[:,2].min()
mean_productB = data[:,2].mean()
relative_range_productB = (max_productB - min_productB) / mean_productB*100

print('Relative range for Product B:', relative_range_productB)

Relative range for Product B: 18.30065359477124


### which means the range of sales data is 18.3% of the mean sales for product B

### 3.Determine the relative standard deviation for Product A. Interpret the result and discuss its implications for understanding the variability relative to the mean.

In [40]:
mean_productA = data[:,1].mean()
std_productA = data[:,1].std()
coefficient_of_variation_productA = (std_productA / mean_productA)*100

print('relative standard deviation for Product A:', coefficient_of_variation_productA)

relative standard deviation for Product A: 7.838035458037848
